src/fs/types.rs

```diff
@@ -0,0 +1,212 @@
+use std::{hash::{Hash, Hasher}, path::{Path, PathBuf}};
+use smash_arc::{Hash40, Region};
+use std::ops::Deref;
+
+#[derive(Debug)]
+pub enum RejectionReason {
+    DuplicateFile(PathBuf),
+    NotFound(SmashPath),
+    MissingExtension
```

Comment:
These variants honestly aren't 100% clear to me, mind adding a doc comment explaining at least the first 2? Just a few words, as I think "duplicate" and "not found" can be unclear as we're ultimately working with multiple filesystems here (the SD filesystem, the arc filesystem, romfs).

Just something like clarifying X is a duplicate clashing with Y, "not found" in where, etc.